internal/run/convert_test.go

```diff
@@ -40,3 +46,148 @@ func TestV3ReleaseFound(t *testing.T) {
+					"VERSION": "2",
+				},
+			},
+		},
+		&corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "other.v1",
+				Namespace: "example",
+				Labels: map[string]string{
+					"NAME":    "other",
+					"OWNER":   "TILLER",
+					"STATUS":  "DEPLOYED",
+					"VERSION": "1",
+				},
+			},
+		},
+	)
+
+	cmaps, err := getReleaseConfigmaps(clientset, "myapp", "example", "")
+	assert.NoError(t, err)
+	assert.Equal(t, len(cmaps.Items), 2)
+}
+
+func TestPreserveV2(t *testing.T) {
+
+	clientset := fake.NewSimpleClientset(
+		&corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "myapp.v1",
+				Namespace: "example",
+				Labels: map[string]string{
+					"NAME":    "myapp",
+					"OWNER":   "TILLER",
+					"STATUS":  "DEPLOYED",
+					"VERSION": "1",
+				},
+			},
+		},
+		&corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "myapp.v2",
+				Namespace: "example",
+				Labels: map[string]string{
+					"NAME":    "myapp",
+					"OWNER":   "TILLER",
+					"STATUS":  "DEPLOYED",
+					"VERSION": "2",
+				},
+			},
+		},
+		&corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "other.v1",
+				Namespace: "example",
+				Labels: map[string]string{
+					"NAME":    "other",
+					"OWNER":   "TILLER",
+					"STATUS":  "DEPLOYED",
+					"VERSION": "1",
+				},
+			},
+		},
+	)
+
+	clientset.Fake.AddReactor("update", "configmap", func(action k8stesting.Action) (handled bool, ret runtime.Object, err error) {
+		configmapList := &corev1.ConfigMapList{
+			Items: []corev1.ConfigMap{
+				corev1.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "myapp.v1",
+						Namespace: "example",
+						Labels: map[string]string{
+							"NAME":    "myapp",
+							"OWNER":   "none",
+							"STATUS":  "DEPLOYED",
+							"VERSION": "1",
+						},
+					},
+				},
+				corev1.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "myapp.v2",
+						Namespace: "example",
+						Labels: map[string]string{
+							"NAME":    "myapp",
+							"OWNER":   "none",
+							"STATUS":  "DEPLOYED",
+							"VERSION": "2",
+						},
+					},
+				},
+			},
+		}
+
+		return true, configmapList, nil
+	})
+
+	opts := convertcmd.ConvertOptions{
+		DeleteRelease:    false,
+		DryRun:           false,
+		ReleaseName:      "myapp",
+		StorageType:      "configmap",
+		TillerNamespace:  "example",
+		TillerOutCluster: false,
+	}
+
+	err := preserveV2(clientset, opts)
+	assert.NoError(t, err)
+
+	myappv1, err := clientset.CoreV1().ConfigMaps("example").Get("myapp.v1", metav1.GetOptions{})
+	assert.NoError(t, err)
+	assert.Equal(t, myappv1.Labels["OWNER"], "none")
+
+	myappv2, err := clientset.CoreV1().ConfigMaps("example").Get("myapp.v2", metav1.GetOptions{})
+	assert.NoError(t, err)
+	assert.Equal(t, myappv2.Labels["OWNER"], "none")
+
+	other, err := clientset.CoreV1().ConfigMaps("example").Get("other.v1", metav1.GetOptions{})
+	assert.NoError(t, err)
+	assert.NotEqual(t, other.Labels["OWNER"], "none")
```

Comment:
For clarity, a table driven test would be a refactor against this part.  Not the fixtures themselves.

Reply:
ok for this part it probably makes sense

Reply:
done